### PR TITLE
feat(api-tiers): add db-first tier resolution with env fallback

### PR DESCRIPTION
**Pull request by Katsiarynakavaleuskaya**
## Summary
- implement DB-first API tier resolution in `app/middleware/api_tiers.py` when `SUBSCRIPTION_DB_ENABLED=true`
- enforce fail-closed behavior for DB error/invalid-tier outcomes (no env fallback in those cases)
- keep env fallback only for explicit DB miss to preserve migration compatibility
- add focused tests for DB-enabled, DB-miss, DB-error, and invalid-tier paths; document policy in `app/AGENTS.md`

## Scope
- `app/middleware/api_tiers.py`
- `tests/test_api_tiers.py`
- `app/AGENTS.md`
- `docs/audit/PR_XXX_API_TIERS_DB_LOOKUP_AUDIT.md`

## Verification Evidence
- `pre-commit run --all-files` ✅
- `pytest -q tests/test_api_tiers.py` ✅
- `make verify` ✅
- `diff-cover --fail-under=97` via `make verify`: **100%** ✅

## DoD
- [x] `SUBSCRIPTION_DB_ENABLED=true` uses DB lookup
- [x] DB miss uses env fallback
- [x] DB error/invalid tier fail closed (no env fallback for access checks)
- [x] unknown/unresolved key remains denied by tier guards
- [x] tests cover db-enabled/db-miss/db-error/invalid-tier paths
- [x] `make verify` green and diff-cover >=97%
- [x] AGENTS policy updated

## Security Notes
- DB query is parameterized and fixed-schema (no dynamic identifier interpolation)
- DB error/invalid-tier outcomes are fail-closed for guard checks
- raw API keys are not logged in tier resolution events

## Discussion Thread Pass
- [x] Discussion-thread pass completed
- [x] Fixed in commit mapping completed

### Fixed in Commit Mapping
- No actionable review comments

## Deferred / Follow-ups
- next PR: OpenAPI contract hygiene for `/api/v1/export/sign` (typed response + regenerated frontend artifacts)

## Summary by CodeRabbit

* **New Features**
  * DB-backed subscription tier resolution with environment-based fallback when DB is disabled or misses.
* **Bug Fixes / Security**
  * Fail-closed behavior: deny access on DB errors or invalid tier data to avoid accidental privilege escalation.
* **Documentation**
  * Added audit/gating docs clarifying DB-first policy, failure modes, and acceptance criteria.
* **Tests**
  * Expanded tests covering DB-enabled/disabled paths, fallback rules, and edge cases.
* **Chores**
  * Updated baseline metadata/timestamp entries.

### CODE REVIEW

File: tests/test_api_tiers.py [hunk 1]
```diff
@@ -66,29 +67,134 @@ def test_anonymous_allowed_in_dev(self) -> None:
         assert _validate_api_key_tier("any_random_key", SubscriptionTier.VIP) is True
 
     @patch.dict(
-        os.environ, {"APP_ENV": "production", "DEBUG": "false", "SUBSCRIPTION_DB_ENABLED": "false"}
+        os.environ,
+        {
+            "APP_ENV": "production",
```
[the remark this answers is not in the document]
Comment (Katsiarynakavaleuskaya) [reply tests/test_api_tiers.py]: Addressed in 3611efe: tests for PRO_API_KEYS and comma-separated env handling added (test_production_mode_without_db_resolves_pro_api_keys_csv, test_production_db_disabled_uses_env_detection, and env fallback coverage).

File: app/middleware/api_tiers.py [hunk 2]
```diff
@@ -66,6 +67,76 @@ class SubscriptionTier(str, Enum):
...
+        logger.warning("Subscription DB lookup returned unknown tier value; using env fallback")
+    return parsed_tier
+
+
+def _resolve_tier_from_env(api_key: str) -> SubscriptionTier | None:
+    """Resolve tier via environment/test-key fallback."""
+    if api_key == TEST_KEY_VIP:
+        return SubscriptionTier.VIP
+    if api_key == TEST_KEY_PRO:
+        return SubscriptionTier.PRO
```
[the remark this answers is not in the document]
Comment (Katsiarynakavaleuskaya) [reply app/middleware/api_tiers.py]: Addressed in 0f150e2 / 3611efe: test keys restricted to non-production via allow_test_keys=not is_production in _resolve_tier_from_env (and call sites).
